dev/com.ibm.ws.classloading/src/com/ibm/ws/classloading/internal/ParentLastClassLoader.java

```diff
@@ -84,17 +85,28 @@ public URL getResource(String resName) {
                     // first check our classpath
                     rc = findClass(className);
```

Comment:
Technically this could be changed to call a different method that didn't throw the exception similar to what you did in the GatewayClassLoader.

Reply:
I suppose this is true.  My approach here was to reuse the generated exception from findClass when the parent returned null.  If we did what you suggest then we would have to generate an exception out of the parent delegation.  At that point there is probably little value in having NoClassNotFoundLoader parent, or at least there is little value in calling parent.loadClassNoException here.  But you are correct that it would save us one CNFE for searching self before parent when the class exists in parent.

Reply:
Yeah as you suggest, this can be done in a separate PR after this one.  An example where this would improve things is in the case that it is using liberty API / SPI or spec APIs in the application.  This is outside the scope of the PR and can be done in the follow on one.